### DISCUSSION TITLE
Possible to get separate activation maps for each detected object?

### DISCUSSION THREAD

**Discussion by dannn-tran** [jacobgil/pytorch-grad-cam#272] [Q&A]
Hey!

I followed the tutorial [here](https://github.com/jacobgil/pytorch-grad-cam/blob/master/tutorials/Class%20Activation%20Maps%20for%20Object%20Detection%20With%20Faster%20RCNN.ipynb) and got quite cool results with my own image :). My image has two people, and the model correctly detects the bounding boxes and labels. When I visualise the activation map, I realise that it is a single map for all detected objects. I thus wonder if it is possible to obtain separate activation map for each detected object.

Would really appreciate any insight on this. Thanks!

**Comment by jacobgil** [comment]
Hi picasa,
The tutorial has a section called "Optional: Renormalizing the CAMs inside every bounding box".
Can you give it a try?

**Reply by dannn-tran** [reply]
Hi Jacob,
Thanks for the reply. I did try normalising the CAMs within each bounding box but the results are not ideal when there is a lot of overlapping between the boxes. Recently I learnt about DETR and I think it might be better suited to accomplish what I want. Thanks regardless!
